sys/net/network_layer/ng_icmpv6/echo/ng_icmpv6_echo.c

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (C) 2015 Martine Lenders <mlenders@inf.fu-berlin.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ */
+
+#include "net/ng_netbase.h"
+
+#include "net/ng_icmpv6.h"
+#include "net/ng_icmpv6/echo.h"
+#include "utlist.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+void ng_icmpv6_echo_req_handle(kernel_pid_t iface, ng_ipv6_hdr_t *ipv6_hdr,
+                               ng_icmpv6_echo_t *echo, uint16_t len)
+{
+    uint8_t *payload = (uint8_t *)echo + 1;
```

Comment:
No that is how pointer arithmics work, the units of bytes you add are of the magnitude of the type the pointer refers to (so I add sizeof(ng_icmpv6_echo_t) byte here). But it might be that a pair of paranthesis is missing, so the cast is happening before the addition.